### ISSUE TITLE
Sapper Integration? (Errors)

### ISSUE THREAD

**Issue by WeAreELIC** [YogliB/svelte-fullcalendar#348]
Hey, do you have any examples integrating FullCalendar with Sapper?

I'm attempting to do so now but I'm getting the following error:

```
.../node_modules/@fullcalendar/core/main.js:108
    var matchesMethod = Element.prototype.matches ||

ReferenceError: Element is not defined
  at .../node_modules/@fullcalendar/core/main.js:108:25
  at .../node_modules/@fullcalendar/core/main.js:8:68
  at Object.<anonymous> (.../node_modules/@fullcalendar/core/main.js:11:2)
  at Module._compile (internal/modules/cjs/loader.js:1151:30)
  at Object.Module._extensions..js (internal/modules/cjs/loader.js:1171:10)
  at Module.load (internal/modules/cjs/loader.js:1000:32)
  at Function.Module._load (internal/modules/cjs/loader.js:899:14)
  at Module.require (internal/modules/cjs/loader.js:1040:19)
  at require (internal/modules/cjs/helpers.js:72:18)
  at .../node_modules/@fullcalendar/interaction/main.js:8:85
```
Thanks!

**Comment by YogliB** [comment]
I'll add a Sapper example when I release the new version (FC v5).

**Comment by dasper** [comment]
This is an issue with server side rendering in Sapper rather than with the package itself. There is some sapper documentation which can hopefully help. 

https://sapper.svelte.dev/docs#Making_a_component_SSR_compatible

any import which immediately tries to use the DOM will fail when it runs server-side. Using a dynamic import (await import('stuff')) is the solution for this. Or to attempt to use `onMount` like this:
``` javascript
<script>
	import { onMount } from 'svelte';

	let MyComponent;

	onMount(async () => {
		const module = await import('my-non-ssr-component');
		MyComponent = module.default;
	});
</script>
```

**Comment by YogliB** [comment]
@dasper 
I didn't quite understand if there's something I can do.

BTW, I'm currently holding off on all updates until I'll manage to use v5 of FullCalendar with Svelte.

**Comment by dasper** [comment]
Hello @YogliB, I am not sure because I am not a strong front end or JS developer. I just ran into this similar issue and knew the error was because the way FullCalendar was being instantiated it is expecting DOM elements but Sapper's template has not generated them yet. One fix is using `onMount` to then use the FullCalendar library as you would on any traditional static site. 

Perhaps you can add some logic before bootstrapping the code that utilizes DOM elements; something like `const isSSR = typeof window === 'undefined';` so it knows it cannot access the elements through window or document. Or intercept the way it finds the element and inject it yourself. Beyond that, I am too ignorant of Node and the underlying workings to be of any help.

**Comment by klevente** [comment]
Hello, I managed to use the component with Sapper, based on what @dasper suggested. It is a bit hacky, but works well:
1. First create a placeholder component, that is completely empty, call it `Placeholder`
2. In the component/route where you want to use the calendar, import it dynamically along with any plugins you need:
```javascript
let FullCalendar;
let dayGridPlugin;

let selected = Placeholder;

onMount(async () => {
    const module = await import('svelte-fullcalendar');
    FullCalendar = module.default;
    const dayModule = await import('@fullcalendar/daygrid');
    dayGridPlugin = dayModule.default;
    selected = FullCalendar;
});
```
4. In the markup, add a `svelte:component` that will switch out the placeholder for the actual calendar when it loads:
```html
<svelte:component this="{selected}" plugins="{[ dayGridPlugin ]}" />
```

**Comment by YogliB** [comment]
solved on v1.0.0
